content/docs/cucumber/api.md

```diff
@@ -596,68 +596,151 @@ See more documentation on [tags](#tags).
 
 ## Global hooks
 
-{{% block "ruby" %}}
-A global hook will run once before any scenario is run. Put the
-code at the top-level in your `env.rb` file (or any other file under
-`features/support` directory).
+Global hooks will run once before any scenario is run or after all scenario have
+been run. Put the code at the top-level in your `env.rb` file (or any other file
+under `features/support` directory).
 
-Use `Kernel#at_exit` for global teardown.
+### BeforeAll
 
-Example:
-{{% /block %}}
-```ruby
-my_heavy_object = HeavyObject.new
-my_heavy_object.do_it
+`BeforeAll` run before any scenario is run.
 
-at_exit do
-  my_heavy_object.undo_it
+{{%block "ruby" %}}
+```ruby
+BeforeAll do
+  # Do something before any scehario is executed
 end
 ```
+{{% /block %}}
 
-{{% block "java,kotlin,scala" %}}Cucumber-JVM does not support global hooks.{{% /block %}}
-{{% block "javascript" %}}Cucumber.js does not support global hooks.{{% /block %}}
+{{%block "javascript" %}}
+```javascript
+const { BeforeAll } = require('@cucumber/cucumber');
 
-## Running a hook only once
+// Synchronous
+BeforeAll(function () {
+  // perform some shared setup
+});
 
-{{% text "ruby" %}}
-If you have a hook you only want to run once, use a global variable:
+// Asynchronous Callback
+BeforeAll(function (callback) {
+  // perform some shared setup
 
+  // execute the callback (optionally passing an error when done)
+});
+```
+{{% /block %}}
 
+{{%block "java" %}}
+```java
+@BeforeAll
+public static void beforeAll() {
```

Comment:
Not yet available in Cucumber JVM v6. Will be in V7.

Reply:
I though v7 had been released 😅

But actually you have spotted an issue here: the documentation do not care about versions of cucumber. Someone with an older version of cucumber will ever find the appropriate documentation.

Reply:
That's always been like that and usually not a big problem.

Reply:
I did want to release v7. Just haven't gotten around to it yet. :laughing:

Reply:
So, we'll see at the moment of merging that PR, if v7 not yet released, we'll remove the java part, replacing it with "coming soon"

Reply:
What about simply redirecting such technical documentation to the repositories like [cucumber-js](https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/hooks.md), and [cucumber-ruby](https://github.com/cucumber/cucumber-ruby/blob/main/features/docs/writing_support_code/hooks/README.md)?

Those remain tech documentations: there is no need to keep those away from the code as opposed to gherkin documentation and references, guides, and other content on cucumber.io/docs

It will be versioned: users will have the possibility to find the appropriate documentation to their versions

It will be easier for us, as developers, to keep it up-to-date and to make it part of our development processes (I mean, like we keep the changelog up-to-date as part of our PR and reviewers take care of this, we could do the same with such tech documentations)

What do you think?

Reply:
For my part I think the technical docs on this site should be super thin (closer to concepts than specific APIs) and hand off to relevant docs in the implementation repos. That might just be me though...

Reply:
yes, I have the same though

Reply:
@mpkorstanje should I revert the changes for Java despite the release of cucumber-jvm 7 RC1?

Reply:
Probably best for now. Maybe split them out in to a separate pull request.

Reply:
👍 
Will do before merging if cucumber-jvm 7 still not released.

Won't merge this now as cucumber-ruby still need to be released too. But I think we may release it pretty soon.